### PR TITLE
[Feature] Integrate some simple trace into be HTTP service

### PR DESCRIPTION
**Pull request by xlwh**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by xlwh** [comment]
run starrocks_clang-format

File: be/src/common/config.h [hunk 1]
```diff
@@ -715,6 +715,11 @@ CONF_Bool(enable_segment_overflow_read_chunk, "true");
 
 CONF_Int32(max_batch_publish_latency_ms, "100");
 
+// Some config for opentelemetry tracing.
+CONF_Bool(enable_tracer, "false");
+CONF_String(jaeger_server_endpoint, "localhost");
+CONF_Int64(jaeger_server_port, "6381");
```
Comment (decster) [inline be/src/common/config.h:721]: How about combining them into one config, just `jaeger_endpoint`, if it is empty(default) just disable tracer.

Comment (xlwh) [reply be/src/common/config.h:721]: OK, good idea!

File: be/src/common/tracer.cpp [hunk 2]
```diff
@@ -4,35 +4,56 @@
...
 
+bool Tracer::is_enabled() const {
+    return !config::jaeger_endpoint.empty();
+}
+
 Span Tracer::start_trace(const std::string& trace_name) {
+    if (!is_enabled()) {
+        return noop_tracer->StartSpan(trace_name);
+    }
     return _tracer->StartSpan(trace_name);
```
Comment (decster) [inline be/src/common/tracer.cpp:57]: why not cache noop_tracer & _tracer to _tracer, so you won't check is_enabled

File: be/src/common/tracer.cpp [hunk 3]
```diff
@@ -4,34 +4,50 @@
...
-    _tracer = provider->GetTracer(service_name, OPENTELEMETRY_SDK_VERSION);
+    if (!config::jaeger_endpoint.empty()) {
+        opentelemetry::exporter::jaeger::JaegerExporterOptions opts;
+        opts.endpoint = config::jaeger_endpoint;
+        auto jaeger_exporter = std::unique_ptr<opentelemetry::sdk::trace::SpanExporter>(
+                new opentelemetry::exporter::jaeger::JaegerExporter(opts));
+        auto processor = std::unique_ptr<opentelemetry::sdk::trace::SpanProcessor>(
+                new opentelemetry::sdk::trace::SimpleSpanProcessor(std::move(jaeger_exporter)));
+        const auto jaeger_resource = opentelemetry::sdk::resource::Resource::Create(
+                std::move(opentelemetry::sdk::resource::ResourceAttributes{{"service.name", service_name}}));
```
[the remark this answers is not in the document]
Comment (decster) [reply be/src/common/tracer.cpp:34]: `std::move(opentelemetry::sdk::resource::ResourceAttributes{{"service.name", service_name}})`

[the remark this answers is not in the document]
Comment (xlwh) [reply be/src/common/tracer.cpp]: OK, thanks!

**Review by decster — APPROVED** [review]

**Comment by decster** [comment]
run starrocks_be_unittest

**Review by chaoyli — APPROVED** [review]

**Pull request merged**